doc/sphinx/src/development.rst

```diff
@@ -62,6 +62,34 @@ parallelism interface that is needed for managing memory cached in
 tightly nested loops. The wrappers are documented
 :ref:`here <nested par for>`.
 
+View of Views
+-------------
+
+Special care needs to be taken when working with a ``View`` of ``View``.
+
+To repeat the Kokkos documenation: `Don't use them <https://kokkos.org/kokkos-core-wiki/ProgrammingGuide/View.html#can-i-make-a-view-of-views>`__
+
+But if you have to (which is the case in some places inside Parthenon)
+then follow this pattern:
```

Comment:
Why is this preferred over just making the inner `View`s unmanaged in a `View` of `View`s? My understanding is that the unmanaged view doesn't call the destructors, so doing this would also solve the problem since I don't think we ever make a view of views that actually manages the inner views (maybe this is wrong?). I think it would be trivial to make an `UnmanagedParArray#D` by just changing the view type it is templated on. Then we wouldn't have to have raw Kokkos views floating around in the code.

Reply:
Honestly, I don't know and potentially am afraid of some unknown side effects.
What happens to the reference counter of the existing managed view when we assign it to an unmanaged element of the outer view.
Similarly, for the boundary info and prolong/restrict info outer view that contain objects which implicitly contain `ParArray#D`s. If they're not reference counted, we'd manually have to destruct those, don't we (or how else would the inner object know that they don't need to exist anymore)?

Reply:
> “Unmanaged” means that Kokkos does not do reference counting or automatic deallocation for those Views. 

I think this implies that the unmanaged view doesn't increment the reference count. So it is possible that the other view pointing to the same memory deallocates the memory while the unmanaged view still points to it, but the unmanaged view will never try to call the destructor.

I believe that in `BndInfo` and prolong/restrict info we are never creating a `ParArray#D` that is only held by those objects. Rather, we are essentially "pointing" to `ParArray#D`s that are held elsewhere (e.g. the communication buffer, variable data). I think the other places those objects persist longer than the packs, `BndInfo`, etc. so it is safe to have the internal Views in views of views unmanaged.

I think my real concern is that it is unclear (at least to me) what `SequentialHostInit` is doing.

Reply:
> From the Kokkos docs:
> 
> > “Unmanaged” means that Kokkos does not do reference counting or automatic deallocation for those Views.
> 
> I think this implies that the unmanaged view doesn't increment the reference count. So it is possible that the other view pointing to the same memory deallocates the memory while the unmanaged view still points to it, but the unmanaged view will never try to call the destructor.
> 
> I believe that in `BndInfo` and prolong/restrict info we are never creating a `ParArray#D` that is only held by those objects. Rather, we are essentially "pointing" to `ParArray#D`s that are held elsewhere (e.g. the communication buffer, variable data). I think the other places those objects persist longer than the packs, `BndInfo`, etc. so it is safe to have the internal Views in views of views unmanaged.
> 
> I think my real concern is that it is unclear (at least to me) what `SequentialHostInit` is doing.

The PR that added this feature to Kokkos is just a few lines (https://github.com/kokkos/kokkos/pull/7229 excluding the test code).
AFAIK the property effectively tells Kokkos to explicitly call the destructor (line 229 https://github.com/kokkos/kokkos/pull/7229/files#diff-0d719ff2418eb0512b065dca1765d2788ddd4a524734f8f96b7af34a22f4b560) which results in a clean deallocation of the inner view.
Otherwise, the default dtor of the outer view would call the dtor of the inner views in a parallel region, which is illegal.

But maybe I'm also wrong.

But if that's correct, then I somehow feel more comfortable with this way (for the moment) as I understand that concept whereas it's unclear to me if there are any side effects (not necessarily from a Kokkos point of view but from the Parthenon usage point of view) in using unmanaged inner views.

Reply:
I think what I didn't understand was that it seemed like I could only set `SequentialHostInit` for host space views of device space views and it seemed like we actually needed this for device space views of device space views. But it is very possible that I just don't understand what is going on.